kratos/utilities/variable_utils.h

```diff
@@ -268,6 +268,21 @@ class KRATOS_API(KRATOS_CORE) VariableUtils
         KRATOS_CATCH("")
     }
 
+    /**
+     * @brief Sets the nodal value of any variable to zero
+     * @param rVariable reference to the scalar variable to be set
+     * @param rNodes reference to the objective node set
+     */
+    template< class TType >
+    void SetVariableToZero(
+        const Variable< TType >& rVariable,
+        NodesContainerType& rNodes)
```

Comment:
Can you template this too? Then we can also use it for Elements, Conditions and MasterSlaveConstraints

Reply:
I was thinking about doing it. My only concern is that this assumes that you are setting a nodal historical value to zero. If we generalize to other types of entities this is not the case anymore.
Having this in mind, I see two options:

- We add a template argument for the container type and the conditions and elements specialization does a `SetValue` to zero, while the nodal one does a `FastGetSolutionStepValue`. This has two problems, one is that we will need to specialize for all the variable types since we cannot do partial template specialization. Besides, it is not consistent if one wants to set a non-historical variable in the nodes.
- We keep this one as it is and we create a `SetNonHistoricalVariableToZero` which has an extra template argument for the container type and always does a `SetValue`. I think that @loumalouomega already did something similar to set non-historical variables, so probably we can reuse this ones as I did with the `SetVariable` one.

What do you think?

Reply:
(I'm un favor of the second one)

Reply:
I would go for a third option:
Call the method that sets the Nodal-historical values `SetNodalHistoricalVariableToZero` and make it accept only Nodes
Then you can call the other methods `SetVariableToZero` and template it with the containertype

IMO this is the most consistent and self-explaining solution

the current version (`SetVariable`) is misleading since it sets Historical values!

Reply:
That's true, right now can be easily misunderstood. I'll update. @loumalouomega makes sense for you as well?